pkg/controllers/pushsecret/pushsecret_controller.go

```diff
@@ -289,11 +291,20 @@ func (r *Reconciler) handlePushSecretDataForStore(ctx context.Context, ps esapi.
 		Name: storeName,
 		Kind: refKind,
 	}
+	originalSecretData := secret.Data
+	defer func() {
+		secret.Data = originalSecretData
+	}()
```

Comment:
I'm not sure you need to do this as the secret isn't altered in the end even though it's being passed in as a pointer.

The pushSecret operation doesn't touch the original secret.

I was maybe thinking that the loop works on the modified secret, but then that would mean that the restoration should have happened in the loop, but this is not the case.

So to sum up, you can remove this indirection with original data. The secret alternation should have no effect on any further operations.

Reply:
thanks for taking time for explaining it, I was indeed not sure if it was need or not and decide to play safe, I have removed it now.